### PR TITLE
Ccp 1386 ext slider widget

### PR DESCRIPTION
**Pull request by Phmank**
CCP-1386 created first version

### CODE REVIEW

File: frontend/package.json [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+    "prop-types": "^15.6.0",
+    "react": "^16.2.0",
+    "react-dom": "^16.2.0",
+    "react-hot-loader": "4.0.1",
+    "react-transform-catch-errors": "^1.0.2",
+    "redbox-react": "^1.5.0"
+  },
+  "dependencies": {
+    "@shopgate-ps/pwa-extension-kit": "^0.6.0"
+  }
```
Comment (Carsten89) [inline frontend/package.json:43]: please add @shopgate/pwa-common as a peer dependency

File: frontend/components/Slider/index.jsx [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+        loop
+      >
+        {slides}
+      </Slider>
+    </div>
+  );
+};
+
+SliderWidget.propTypes = {
+  settings: PropTypes.shape({
```
Comment (Carsten89) [inline frontend/components/Slider/index.jsx:101]: Please mark settings and slides as required here

File: frontend/components/Slider/index.jsx [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+    slide => createSlide(slide)
+  ));
+
+  return (
+    <div>
+      <div className={styles.headlineContainer}>
+        <h3 className={styles.headline}>{props.settings.headline ? props.settings.headline : ''}</h3>
+      </div>
+      <Slider
+        autoPlay
```
Comment (Carsten89) [inline frontend/components/Slider/index.jsx]: autoPlay and loop should be a setting too (like headline), but with defaults

File: frontend/components/Slider/index.jsx [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+    </div>
+  );
+
+  const slides = props.settings.slides.map((
+    slide => createSlide(slide)
+  ));
+
+  return (
+    <div>
+      <div className={styles.headlineContainer}>
```
Comment (Carsten89) [inline frontend/components/Slider/index.jsx:87]: this container should not be rendered if there is no headline

File: frontend/components/Slider/index.jsx [hunk 5]
```diff
@@ -0,0 +1,88 @@
...
+        <div className={styles.sliderWrapper}>
+          {slide.map((
+            slideItem => createSliderItem(slideItem)
+          ))}
+        </div>
+      </Slider.Item>
+    </div>
+  );
+
+  const slides = props.settings.slides.map((
```
Comment (Carsten89) [inline frontend/components/Slider/index.jsx]: this function should be moved out of the "SliderWidget" func

File: frontend/components/Slider/index.jsx [hunk 6]
```diff
@@ -0,0 +1,88 @@
...
+      </Link>
+    );
+  };
+
+  /**
+  * Creates an Slide with Items.
+  * @param {Object} slide The Slide data.
+  * @return {JSX} The rendered Slide.
+  */
+  const createSlide = slide => (
```
Comment (Carsten89) [inline frontend/components/Slider/index.jsx]: this function should be moved out of the "SliderWidget" func

File: frontend/components/Slider/index.jsx [hunk 7]
```diff
@@ -0,0 +1,88 @@
...
+ * @param {Object} props The component props.
+ * @returns {JSX}
+ */
+const SliderWidget = (props) => {
+  /**
+  * Creates an Item for a Slide.
+  * @param {Object} slideItem The Slide Item data.
+  * @return {JSX} The Slide Item.
+  */
+  const createSliderItem = (slideItem) => {
```
Comment (Carsten89) [inline frontend/components/Slider/index.jsx]: this function should be moved out of the "SliderWidget" func

File: frontend/components/Slider/index.jsx [hunk 8]
```diff
@@ -0,0 +1,88 @@
...
+        </div>
+      </Slider.Item>
+    </div>
+  );
+
+  const slides = props.settings.slides.map((
+    slide => createSlide(slide)
+  ));
+
+  return (
```
Comment (Carsten89) [inline frontend/components/Slider/index.jsx:85]: if props.settings.slides is empty it should render nothing here

File: frontend/components/Slider/index.jsx [hunk 9]
```diff
@@ -0,0 +1,88 @@
...
+    );
+  };
+
+  /**
+  * Creates an Slide with Items.
+  * @param {Object} slide The Slide data.
+  * @return {JSX} The rendered Slide.
+  */
+  const createSlide = slide => (
+    <div key="">
```
Comment (Carsten89) [inline frontend/components/Slider/index.jsx]: why is this div needed?

File: frontend/components/Slider/index.jsx [hunk 10]
```diff
@@ -0,0 +1,88 @@
...
+ * @returns {JSX}
+ */
+const SliderWidget = (props) => {
+  /**
+  * Creates an Item for a Slide.
+  * @param {Object} slideItem The Slide Item data.
+  * @return {JSX} The Slide Item.
+  */
+  const createSliderItem = (slideItem) => {
+    if (slideItem.img === '' || slideItem.img === undefined) {
```
Comment (Carsten89) [inline frontend/components/Slider/index.jsx]: check for typeof object or string here and add comments to the different ifs

File: frontend/components/Slider/index.jsx [hunk 11]
```diff
@@ -0,0 +1,88 @@
...
+import Link from '@shopgate/pwa-common/components/Link';
+import Slider from '@shopgate/pwa-common/components/Slider';
+import styles from './style';
+
+/**
+ * Renders the component.
+ * @param {Object} props The component props.
+ * @returns {JSX}
+ */
+const SliderWidget = (props) => {
```
Comment (Carsten89) [inline frontend/components/Slider/index.jsx:56]: key="" is not an option ;)

**Review by Carsten89 — CHANGES_REQUESTED** [review]

File: README.md [hunk 12]
```diff
@@ -1 +1,58 @@
...
+
+## Configuration
+
+If you want to use the image slider widget, you can create a HTML widget with following content:
+
+```json
+<!--Widget
+{
+  "type": "@shopgate/ext-slider-widget/SliderWidget",
+  "settings": {
```
Comment (Carsten89) [inline README.md:15]: please add also examples for the other supported formats

File: frontend/package.json (hunk 1 above)
Comment (Phmank) [reply frontend/package.json:43]: done

File: frontend/components/Slider/style.js [hunk 13]
```diff
@@ -0,0 +1,38 @@
+import { css } from 'glamor';
```
Comment (Carsten89) [inline frontend/components/Slider/style.js:1]: do we have some variables we could use here for the hardcoded numbers?

**Review by Carsten89 — CHANGES_REQUESTED** [review]

File: frontend/components/Slider/index.jsx (hunk 7 above)
Comment (Phmank) [reply frontend/components/Slider/index.jsx]: done

File: frontend/components/Slider/index.jsx (hunk 8 above)
Comment (Phmank) [reply frontend/components/Slider/index.jsx:85]: done

File: frontend/components/Slider/index.jsx (hunk 9 above)
Comment (Phmank) [reply frontend/components/Slider/index.jsx]: removed. done.

File: frontend/components/Slider/index.jsx (hunk 4 above)
Comment (Phmank) [reply frontend/components/Slider/index.jsx:87]: done

File: frontend/components/Slider/index.jsx (hunk 10 above)
Comment (Phmank) [reply frontend/components/Slider/index.jsx]: only check for string needed. done.

File: frontend/components/Slider/index.jsx (hunk 3 above)
Comment (Phmank) [reply frontend/components/Slider/index.jsx]: done

File: frontend/components/Slider/index.jsx (hunk 11 above)
Comment (Phmank) [reply frontend/components/Slider/index.jsx:56]: done

File: README.md (hunk 12 above)
Comment (Phmank) [reply README.md:15]: done

File: frontend/components/Slider/index.jsx [hunk 14]
```diff
@@ -0,0 +1,117 @@
...
+      <img className={styles.item} key={`${slideItem.img}-${index}`} alt="" src={slideItem.img} />
+    );
+  }
+
+  return (
+    <Link
+      href={slideItem.link}
+      key={slideItem.link + index}
+    >
+      <img className={styles.item} key={`${slideItem.img}-${index}`} alt="" src={slideItem.img} />
```
Comment (Carsten89) [inline frontend/components/Slider/index.jsx]: you don't need a key here, since it's the only element in the Link

File: README.md [hunk 15]
```diff
@@ -1 +1,66 @@
...
+
+This extension provides a image slider widget.
+
+The widget is configurable in regards of amount of slides and items on the slides.
+
+## Configuration
+
+If you want to use the image slider widget, you can create a HTML widget with following content:
+
+```json
```
Comment (Carsten89) [inline README.md]: i changed this to html

**Review by Carsten89 — APPROVED** [review]

**Pull request merged**